### DISCUSSION TITLE
Frozen simulation in Performing Initial Setup

### DISCUSSION THREAD

**Discussion by Ali1990dashti** [idaholab/moose#20205] [Q&A General] [answered]
Dear Community,
I am trying to run a simulation for a mesh with about `12 K` nodes and `80-90 K` elements. When I run it on a HPC with about `20` cores the simulation get frozen in `Performing Initial Setup....`. Based on [this discussion](https://groups.google.com/g/moose-users/c/MVdlbM7vnr4), I tried less cores but still the same result. When I also try it in my local virtual machine I face the following error:
```
===
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 7259 RUNNING AT alid-VirtualBox
=   EXIT CODE: 9
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES

```
My mesh has three layers. One of them is very thin and when I try to have more nodes in this thin layer the problem pops up. I can have thicker layers with much more nodes and elements and run simulation without any problems but it seems too much nodes in a  thin part of my mesh are source of the problem. I am trying to simulation a TH model using one dirac point. I very much appreciate if anyone can give a hint. My mesh is also uploaded here.

[mesh.zip](https://github.com/idaholab/moose/files/7985501/mesh.zip)

**Comment by GiudGiud** [comment, marked as answer]
Hello

So on your local virtual machine, the exit code 9 is most likely because you ran out of memory. To not run out of memory, you can first try to use a distributed mesh `--distributed-mesh`. Or you can try to use less MPI processes. 

On the cluster, we need to find what is actually taking time. 
First, can you please try setting more verbose output. I think it's done like this, not sure:
```
[Outputs]
  [./screen]
    type = Console
    verbose = true
    time_precision = 6
  [../]
[]
```
Then, are you refining this layer in the MOOSE simulation? How? I dont think the refinement is the problem because there is a console output for that that you would have seen, but it's a commonly slow thing which we can accelerate with certain options.

Finally, is it truly frozen or is there a continuous output of points? `Performing Initial Setup....` should gain a new point every like 5 seconds if it's still doing work

Guillaume

**Reply by Ali1990dashti** [reply]
Dear @GiudGiud 
Firstly, thanks for your time.
I did not get what `--distributed-mesh` does mean? You mean changing my mesh?
I tried using 6 to 1 cores in my local virtual machine but no result.
I tried also the output option you suggested but nothing changed. About refinement, I did it in the mesh genertaor (GMSH). It is a bit strange because I only have problem when the target layers is thin and lots of nodes are whithin this thin layer. I have created much bigger meshes with more nodes and the same input file works.
About `Performing Initial Setup`, it is just adding points.

**Reply by GiudGiud** [reply]
` --distributed-mesh` is a command line option to get the simulation to distribute the mesh among all processes rather than replicate it. It saves memory. 

Your VM could be running out of memory at 1 core already then, which means ` --distributed-mesh` wont help. If you could make the mesh smaller, by removing some refinement probably, then you could debug the problem on your VM.

To identify what is taking so long, the easiest way is to reduce the mesh and profile the code. 
We have instructions here on how to do that: https://mooseframework.inl.gov/application_development/profiling.html

**Reply by Ali1990dashti** [reply]
@GiudGiud 

Thanks again for your time.
My problem is really getting strange. I tried several meshes and it seems if I make the mesh finer just more than a limit in the `res` layer (the target layer), the mesh won't work anymore in MOOSE. I have uploade two meshes and if you have a look at them, the difference between Working and NotWorking meshes is very subtle!
[mesh.zip](https://github.com/idaholab/moose/files/7988656/mesh.zip)

**Reply by GiudGiud** [reply]
Can you upload exodus files? On a github repo too please. I m not supposed to open zip files from the internet on my workstation

**Comment by Ali1990dashti** [comment]
Dear @GiudGiud,

Sorry for my late feedback.
My problem solved just by adding more nodes to the mesh! When the difference between number of nodes in each volume is extreme, it doesnot work. I applied two solutions: reducing number of nodes in target layers or adding nodes to other two layers.
I uploaded files in my repo. You can find it here: https://github.com/Ali1990dashti/Mesh_files. Both meshes are almost the same but `Working `mesh has less nodes in target volume.

**Reply by GiudGiud** [reply]
This mesh has lots of issues. I recommend you rework the meshing to make sure the elements arent as skewed, dont have huge aspect ratios and have reasonable gradient in sizes. There's huge elements touching very small ones. 

Even if we found why the initialization is freezing, the results you would get from the solve would be bad. 
You can either improve the mesh in gmsh or use the free version of cubit https://coreform.com/products/coreform-cubit/free-meshing-software/. That should get you nicer meshes (though limited in size)

**Reply by Ali1990dashti** [reply]
Thanks for your hints and time. As you said, the mesh is causing all these issues.
